frontend/src/api/skills/getAllSkills.ts

```diff
@@ -8,16 +8,17 @@ import { errorSnackbar } from 'constants/snackbarVariant';
 import { QUERY_KEYS } from 'constants/queryKeyConstants';
 import { ISkillsListProps } from 'types/skill';
 
-const useGetAllSkills = (): UseQueryResult<ISkillsListProps[], AxiosError> => {
+const useGetAllSkills = (skillName: string): UseQueryResult<ISkillsListProps[], AxiosError> => {
   const { enqueueSnackbar } = useSnackbar();
   const handleSubmitError = (error: AxiosError) => {
     enqueueSnackbar(error?.response?.data, errorSnackbar);
   };
   return useQuery(
-    [QUERY_KEYS.skills],
+    [QUERY_KEYS.skills, skillName],
     async () => {
       const apiClient = apiClientWrapper();
-      const response = await apiClient.get(API.skills);
+      const skillNameEncoded = encodeURIComponent(skillName);
+      const response = await apiClient.get(`${API.skills}?search=${skillNameEncoded}`);
```

Comment:
maybe better to forward queries like params. https://masteringjs.io/tutorials/axios/get-query-params

Reply:
yes